### PR TITLE
start prank.s

### PR DESCRIPTION
**Pull request by Cofflag**

### CODE REVIEW

**Comment by Cofflag** [comment]
Sorry for being so late(•_•)

File: src/prank.c [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+        if (obj->object->subtype1 == 2)
+        {
+            obj->kirbyAbility = 0xc;
+        }
+        else
+        {
+            if (obj->object->subtype1 == 1)
+            {
+                gRngVal = gRngVal * 0x00196225 + 0x3C6EF35F;
+                if (!((gRngVal >> 0x10) & 3))
```
Comment (jiangzhengwenjz) [inline src/prank.c:57]: `if (!(Rand16() & 3))`

File: src/prank.c [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+                {
+                    obj->kirbyAbility = 0xc;
```
Comment (jiangzhengwenjz) [inline src/prank.c:59]:
```suggestion
                    obj->kirbyAbility = KIRBY_ABILITY_COOK;
```

File: src/prank.c [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+        prank->base.flags ^= 1;
+        prank->base.xspeed = -prank->base.xspeed;
+    }
+    if (!(prank->base.unk1 & 7))
+    {
+        u16 temp = 0x2000;
+        s32 x;
+        if (prank->base.flags & 1)
+            temp = -temp;
+        x = prank->base.x + (s16)temp;
```
Comment (jiangzhengwenjz) [inline src/prank.c:108]: I don't see why temp is unsigned

File: src/prank.c [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+        }
+    }
+}
+
+void sub_080A1430(struct Object2 *prank)
+{
+    prank->base.flags |= 4;
+    if (!(prank->base.unk1 & 7))
+    {
+        u16 temp = 0x2000;
```
Comment (jiangzhengwenjz) [inline src/prank.c:136]: same here

File: src/prank.c [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+    {
+        sub_080A1FB8(obj);
+    }
+    obj->unk9E = 0;
+    obj->unk7C = 0;
+    if (obj->object->subtype1 != 0)
+    {
+        if (obj->object->subtype1 == 2)
+        {
+            obj->kirbyAbility = 0xc;
```
Comment (jiangzhengwenjz) [inline src/prank.c:50]:
```suggestion
            obj->kirbyAbility = KIRBY_ABILITY_COOK;
```

**Review by jiangzhengwenjz — CHANGES_REQUESTED** [review]

**Comment by Cofflag** [comment]
https://cexplore.karathan.at/z/OrOsAA
I think this is a fakematch sorry.

File: src/prank.c [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+        {
+            const u32 *a = gUnk_082D88B8;
+            if (a[sub_080023E4(prank->base.unk56, x >> 12, prank->base.y >> 12)] & 0x200)
+            {
+                if (prank->base.unk62 & 4)
+                {
+                    prank->unk83 = 3;
+                    prank->unk78 = sub_080A265C;
+                    prank->base.xspeed = temp1;
+                    prank->base.yspeed = temp1;
```
Comment (jiangzhengwenjz) [inline src/prank.c:246]:
```suggestion
                    prank->base.xspeed = 0;
                    prank->base.yspeed = 0;
```

File: src/prank.c [hunk 7]
```diff
@@ -0,0 +1,257 @@
...
+    }
+    if (prank->base.unk1 == 0x18 || prank->base.unk1 == 0x30) PlaySfx(&prank->base, 139);
+}
+
+void sub_080A170C(struct Object2 *prank)
+{
+    s16 temp1, temp2;
+    prank->base.flags |= 4;
+    temp2 = temp1 = prank->base.unk1 & 7;
+    if (!temp1)
```
Comment (jiangzhengwenjz) [inline src/prank.c:226]:
```suggestion
    prank->base.flags |= 4;
    if (!(prank->base.unk1 & 7))
```

**Review by jiangzhengwenjz — CHANGES_REQUESTED** [review]

File: src/prank.c [hunk 8]
```diff
@@ -0,0 +1,279 @@
...
+                    prank->unk83 = 3;
+                    prank->unk78 = sub_080A265C;
+                    prank->base.xspeed = 0;
+                    prank->base.yspeed = 0;
+                }
+                return;
+            }
+        }
+    }
+   if (prank->base.unk62 & 1)
```
Comment (jiangzhengwenjz) [inline src/prank.c:253]:
```suggestion
    if (prank->base.unk62 & 1)
```

**Review by jiangzhengwenjz — CHANGES_REQUESTED** [review]

File: src/prank.c [hunk 9]
```diff
@@ -0,0 +1,279 @@
...
+            sub_080A2710(prank);
+            break;
+        }
+    }
+}
+
+void sub_080A158C(struct Object2 *prank)
+{
+    ObjectSetFunc(prank, 0xe, sub_080A163C);
+    prank->base.flags &= -3;
```
Comment (jiangzhengwenjz) [inline src/prank.c:204]:
```suggestion
    prank->base.flags &= ~2;
```

File: src/prank.c [hunk 10]
```diff
@@ -0,0 +1,279 @@
...
+    if (prank->unk83 != 2
+        || kirby->base.base.base.unk0
+        || kirby->hp <= 0
+        || kirby->unkD4 == 0x27
+        || kirby->unkD4 > 0x7A
+        || kirby->unk110
+        || kirby->base.base.base.flags & 0x3800B00)
+        return FALSE;
+    ObjectSetFunc(prank, 0x10, sub_080A1898);
+    prank->base.flags &= -3;
```
Comment (jiangzhengwenjz) [inline src/prank.c:271]:
```suggestion
    prank->base.flags &= ~2;
```

**Review by jiangzhengwenjz — CHANGES_REQUESTED** [review]

**Review by jiangzhengwenjz — APPROVED** [review]

**Comment by jiangzhengwenjz** [comment]
```
1413968 total bytes of code
967016 bytes of code in src (68.3902%)
446952 bytes of code in asm (31.6098%)
```

**Pull request merged**